vllm/lora/punica_wrapper/punica_base.py

```diff
@@ -195,6 +196,11 @@ def _update_base_metadata(
         )
         self._token_lora_indices[:base_indices.shape[0]].copy_(base_indices)
         self._sampler_indices[:sampler_indices.shape[0]].copy_(sampler_indices)
+        if self.is_prompt_logprobs:
+            assert sampler_indices_padded.size(0) == 1
```

Comment:
this is not necessarily true? If you have 2 short prompted requests batched together?

Reply:
I think it's worth fixing this. This PR is getting close. Just this part is a bit hacky.